astropy/utils/tests/test_data.py

```diff
@@ -372,3 +376,37 @@ def test_is_url_in_cache():
 
     download_file(TESTURL, cache=True, show_progress=False)
     assert is_url_in_cache(TESTURL)
+
+
+@contextlib.contextmanager
+def TemporaryTempDir():
+    """Temporarily redirect all temporary files to a temporary directory"""
+    tempdir = tempfile.mkdtemp()
+    old_tempdir = tempfile.tempdir
+    tempfile.tempdir = tempdir
+    try:
+        yield tempdir
+    finally:
+        tempfile.tempdir = old_tempdir
+        shutil.rmtree(tempdir)
+
+
+def test_get_readable_fileobj_cleans_up_temporary_files():
```

Comment:
Note--the `monkeypatch` helper will set `tempfile.tempdir` back to its original value before the test automatically.